euphony/src/main/cpp/core/charset/CharsetAutoSelector.h

```diff
@@ -0,0 +1,16 @@
+#ifndef EUPHONY_CHARSETAUTOSELECTOR_H
+#define EUPHONY_CHARSETAUTOSELECTOR_H
+
+#include "Charset.h"
+
+namespace Euphony {
+
+    class CharsetAutoSelector : public Charset {
```

Comment:
I wonder why you implemented `Charset` interface here?
I couldn't find implemented `encode` & `decode` functions.